test/ERC20RootVault.test.ts

```diff
@@ -0,0 +1,1149 @@
+                                .deposit([value, value], BigNumber.from(1), [])
+                        ).to.be.revertedWith(Exceptions.LIMIT_OVERFLOW);
+                    });
+                });
+            });
+
+            describe("access control:", () => {
+                it("allowed: any address", async () => {
+                    await this.subject
+                        .connect(this.admin)
+                        .addDepositorsToAllowlist([this.deployer.address]);
+                    await withSigner(this.deployer.address, async (signer) => {
+                        await this.weth
+                            .connect(signer)
+                            .approve(
+                                this.subject.address,
+                                BigNumber.from(10001)
+                            );
+                        await this.usdc
+                            .connect(signer)
+                            .approve(
+                                this.subject.address,
+                                BigNumber.from(10001)
+                            );
+                        await expect(
+                            this.subject
+                                .connect(signer)
+                                .deposit(
+                                    [
+                                        BigNumber.from(10001),
+                                        BigNumber.from(10001),
+                                    ],
+                                    BigNumber.from(1),
+                                    []
+                                )
+                        ).to.not.be.reverted;
+                    });
+                });
+            });
+        });
+
+        describe("#withdraw", () => {
+            it("emits Withdraw event", async () => {
+                await this.subject
+                    .connect(this.admin)
+                    .addDepositorsToAllowlist([this.deployer.address]);
+                await this.weth
+                    .connect(this.deployer)
+                    .approve(this.subject.address, BigNumber.from(10001));
+                await this.usdc
+                    .connect(this.deployer)
+                    .approve(this.subject.address, BigNumber.from(10001));
+                await expect(
+                    this.subject
+                        .connect(this.deployer)
+                        .deposit(
+                            [BigNumber.from(10001), BigNumber.from(10001)],
+                            BigNumber.from(0),
+                            []
+                        )
+                ).not.to.be.reverted;
+
+                var to = randomAddress();
+                await expect(
+                    this.subject.withdraw(
+                        to,
+                        BigNumber.from(0),
+                        [BigNumber.from(0), BigNumber.from(0)],
+                        ["0x00", "0x00"]
```

Comment:
What do these 0x00 options for vaults pull methods mean? If you do not want to give any options, just leave empty bytes array

Reply:
aka [[], []]